.github/workflows/build-and-test.yml

```diff
@@ -1,119 +1,70 @@
+  package_tests:
+    name: Build and Test Swift Package ${{ matrix.platform.name }} (${{ matrix.config }})
     uses: StanfordBDHG/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
+    strategy:
+      matrix:
+        config: [Debug, Release]
+        platform:
+          - name: iOS
+            destination: 'platform=iOS Simulator,name=iPhone 16 Pro'
+          - name: macOS
+            destination: 'platform=macOS,arch=arm64'
+          - name: watchOS
+            destination: 'platform=watchOS Simulator,name=Apple Watch Series 10 (46mm)'
+          - name: visionOS
+            destination: 'platform=visionOS Simulator,name=Apple Vision Pro'
+      fail-fast: false
     with:
       runsonlabels: '["macOS", "self-hosted"]'
       scheme: TemplatePackage
-      destination: 'platform=visionOS Simulator,name=Apple Vision Pro'
-      resultBundle: TemplatePackage-visionOS.xcresult
-      artifactname: TemplatePackage-visionOS.xcresult
-  packagetvos:
-    name: Build and Test Swift Package tvOS
-    uses: StanfordBDHG/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
-    with:
-      scheme: TemplatePackage
-      resultBundle: TemplatePackage-tvOS.xcresult
-      destination: 'platform=tvOS Simulator,name=Apple TV 4K (3rd generation)'
-      artifactname: TemplatePackage-tvOS.xcresult
-  packagemacos:
-    name: Build and Test Swift Package macOS
-    uses: StanfordBDHG/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
-    with:
-      scheme: TemplatePackage
-      resultBundle: TemplatePackage-macOS.xcresult
-      destination: 'platform=macOS,arch=arm64'
-      artifactname: TemplatePackage-macOS.xcresult
-  ios:
-    name: Build and Test iOS
-    uses: StanfordBDHG/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
-    with:
-      path: 'Tests/UITests'
-      scheme: TestApp
-      resultBundle: TestApp-iOS.xcresult
-      artifactname: TestApp-iOS.xcresult
-  ipados:
-    name: Build and Test iPadOS
-    uses: StanfordBDHG/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
-    with:
-      path: 'Tests/UITests'
-      scheme: TestApp
-      destination: 'platform=iOS Simulator,name=iPad Air 13-inch (M2)'
-      resultBundle: TestApp-iPadOS.xcresult
-      artifactname: TestApp-iPadOS.xcresult
-  watchos:
-    name: Build and Test watchOS
+      destination: ${{ matrix.platform.destination }}
+      buildConfig: ${{ matrix.config }}
+      resultBundle: ${{ format('TemplatePackage-{0}-{1}.xcresult', matrix.platform.name, matrix.config) }}
+      artifactname: ${{ format('TemplatePackage-{0}-{1}.xcresult', matrix.platform.name, matrix.config) }}
+  ui_tests:
+    name: Build and Test UI Tests ${{ matrix.platform.name }} (${{ matrix.config }})
     uses: StanfordBDHG/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
+    strategy:
+      matrix:
+        config: [Debug, Release]
+        platform:
+          - name: iOS
+            destination: 'platform=iOS Simulator,name=iPhone 16 Pro'
```

Comment:
It would make sense to test on the same platforms as the old workflow; as well as the ones mentioned to demonstrate a complete setup of tests and a wide variety of possible Swift packages.

Reply:
Nice; thank you!